packages/widget/src/stores/form/FormUpdater.tsx

```diff
@@ -50,21 +57,42 @@ export const FormUpdater: React.FC<{
   // Makes widget config options reactive to changes
   // should update userValues when defaultValues updates and includes additional logic for chains
   useEffect(() => {
-    (Object.keys(defaultValues) as FormFieldNames[]).forEach((key) => {
-      if (previousDefaultValues.current[key] !== defaultValues[key]) {
-        const value =
-          defaultValues[key] ||
-          // set the chain to the current user one if it is not present in the config
-          (key === 'fromChain' || key === 'toChain'
-            ? account.chainId || ''
-            : '');
-        setFieldValue(key, value);
-        resetField(key, { defaultValue: value });
-      }
-    });
-    previousDefaultValues.current = defaultValues;
+    // set values from config
+    if (defaultValues.toAddress) {
+      initialiseSendToWallet(true);
+    }
+
+    // TODO: consider using null in the config to specify when a value should be set to its default
+    //  look at using formDefaultValues for this
+    setUserAndDefaultValues(
+      removeEmptyValuesProperties(defaultValues, account.chainId),
+    );
```

Comment:
This was previously using a ref to hold previous default values but risked missing comparison between config set values and user set values via the widgets UI (form state values).

The checks for this now happen against the current user values in the form state

https://github.com/lifinance/widget/blob/fbbd18c4a1e846e0570417b67b12ac47157418bb/packages/widget/src/stores/form/createFormStore.ts#L97-L106

Doing things this way, irrespective of it the form value was last set by user interaction on the widget or via the config, the last attempt to set the value should be the set value in the widget form.